Snakemake/rules/prepare_files_for_tsinfer.smk

```diff
@@ -37,16 +37,52 @@ if noVCFs == 1:
             bcftools index {output.vcf}
             """
 
+rule get_af:
+    input:
+        config['vcfDir'] + "Chr{chromosome}.vcf.gz"
+    output:
+         "Tsinfer/Info{chromosome}.INFO"
+    params:
+        prefix="Tsinfer/Info{chromosome}"
+    #envmodules:
+    #    config['bcftoolsModule']
+    shell:
+        """
+        bcftools +fill-tags {input} -Oz -o {input} -- -t AN,AC,AF
+        vcftools --gzvcf {input} --out {params.prefix} --get-INFO AC --get-INFO AF
+        """
+
+rule get_major:
+    input:
+        rules.get_af.output
+    output:
+        "Tsinfer/Major{chromosome}.txt"
+    shell:
+        """
+        awk '{{if (NR!=1 && $5>0.5) {{print $1"_"$2","$4}} else if (NR!=1 && $5<0.5) {{print $1"_"$2","$3}}}}' {input} > {output}
+        """
+
+rule combine_major_ancestral:
+    input:
+        ancestral=config['ancestralAllele'],
+        major=rules.get_major.output
+    output:
+        "Tsinfer/AncestralMajor{chromosome}.txt"
+    shell:
+        """
+        join -a1 -t ","  -j 1 -o 1.1,1.2,2.2 <(sort -k1,1 {input.major}) <(sort -k1,1 {input.ancestral}) > tmpMA
+        awk -F, '{{if ($3=="") {{print $1,$2}} else {{print $1,$3}}}}' tmpMA > {output}
+        """
 
 rule decompress:
     input:
-        config['vcfDir'] + "Chr{chromosome}.vcf.gz" #This will take
+        vcf = config['vcfDir'] + "Chr{chromosome}.vcf.gz",
```

Comment:
You can define 'config['vcfDir'] as a variable at the beginning - it won't change anything regarding execution but may be better semantically?

Reply:
Done!